### PR TITLE
placement: add api and pd-ctl support for rule group configuration

### PR DESCRIPTION
**Pull request by disksing**
### What problem does this PR solve?
Tracking issue: https://github.com/pingcap/pd/issues/2721

### What is changed and how it works?
- add API support
- add pd-ctl support
- add test

### Check List

Tests
- Unit test

Code changes
- Has HTTP API interfaces change (Don't forget to [add the declarative for API](https://github.com/pingcap/pd/blob/master/docs/development.md#updating-api-documentation))

### Release note
- Support configuring Index and Override properties for rule groups

### CODE REVIEW

File: server/api/router.go [hunk 1]
```diff
@@ -100,6 +100,11 @@ func createRouter(ctx context.Context, prefix string, svr *server.Server) *mux.R
 	clusterRouter.HandleFunc("/config/rule", rulesHandler.Set).Methods("POST")
 	clusterRouter.HandleFunc("/config/rule/{group}/{id}", rulesHandler.Delete).Methods("DELETE")
 
+	clusterRouter.HandleFunc("/config/rule_group/{id}", rulesHandler.GetGroupConfig).Methods("GET")
+	clusterRouter.HandleFunc("/config/rule_group", rulesHandler.SetGroupConfig).Methods("POST")
+	clusterRouter.HandleFunc("/config/rule_group/{id}", rulesHandler.DeleteGroupConfig).Methods("DELETE")
```
Comment (rleungx) [inline server/api/router.go:105]: I think the user could be a little bit confused about `/config/rule/{group}/{id}` and `/config/rule_group/{id}`.

File: tools/pd-ctl/pdctl/command/config_command.go [hunk 2]
```diff
@@ -428,7 +430,27 @@ func NewPlacementRulesCommand() *cobra.Command {
...
+		Use:   "rule-group",
+		Short: "rule group configurations",
+	}
+	ruleGroupShow := &cobra.Command{
+		Use:   "show [id]",
+		Short: "show rule group configuration(s)",
+		Run:   showRuleGroupFunc,
+	}
+	ruleGroupSet := &cobra.Command{
+		Use:   "set {id} {index} {override}",
```
Comment (rleungx) [inline tools/pd-ctl/pdctl/command/config_command.go:443]: What does the `{}` here mean?

File: server/api/router.go (hunk 1 above)
Comment (disksing) [reply server/api/router.go:105]: yes. But I did not come up with a better idea.

File: tools/pd-ctl/pdctl/command/config_command.go (hunk 2 above)
Comment (disksing) [reply tools/pd-ctl/pdctl/command/config_command.go:443]: oh, I guess I should use <> instead.

File: tools/pd-ctl/pdctl/command/config_command.go [hunk 3]
```diff
@@ -534,3 +556,61 @@ func putPlacementRulesFunc(cmd *cobra.Command, args []string) {
...
+}
+
+func updateRuleGroupFunc(cmd *cobra.Command, args []string) {
+	if len(args) != 3 {
+		cmd.Println(cmd.UsageString())
+		return
+	}
+	index, err := strconv.ParseInt(args[1], 10, 64)
+	if err != nil {
+		cmd.Printf("index %s should be a number", args[1])
```
Comment (rleungx) [inline tools/pd-ctl/pdctl/command/config_command.go:586]: Do we need a `\n` here?

File: tools/pd-ctl/pdctl/command/config_command.go [hunk 4]
```diff
@@ -428,7 +430,27 @@ func NewPlacementRulesCommand() *cobra.Command {
 		Run:   putPlacementRulesFunc,
 	}
 	save.Flags().String("in", "rules.json", "the filename contains rules")
-	c.AddCommand(enable, disable, show, load, save)
+	ruleGroup := &cobra.Command{
+		Use:   "rule-group",
+		Short: "rule group configurations",
+	}
+	ruleGroupShow := &cobra.Command{
+		Use:   "show [<id>]",
```
Comment (rleungx) [inline tools/pd-ctl/pdctl/command/config_command.go:438]:
```suggestion
		Use:   "show [id]",
```

File: tools/pd-ctl/pdctl/command/config_command.go (hunk 3 above)
Comment (disksing) [reply tools/pd-ctl/pdctl/command/config_command.go:586]: good idea.

**Review by rleungx — APPROVED** [review]
LGTM

**Pull request merged**